### ISSUE TITLE
Backwards compatible ?

### ISSUE THREAD

**Issue by pimvdnoll** [mendix/FormatString#6]
Is it still not possible to preserve the behavior texts? I use this widget 150+ times in a project and is almost undoable to check every single widget by hand...plus if you press update all widgets accidentily you can start all over..(yes I did :(( )
if you give every attribute the variable string 0, 1, 2 etc and preserve the behaviour text would be nice or maybe with a migration widget (so update this widget in 2 steps?)

thnx in advance!

**Comment by RoelandSalij** [comment]
Unfortunately, that is not possible. This issue isn't related to this widget and not something that can be solved in this widget. 
Please file a ticket for this issue at support.mendix.com as the issue you describe is related to the platform.

**Comment by pimvdnoll** [comment]
thnx for the quick response but I don't believe this. A new version of a, for instance Graph widget is not breaking existing configuration? How is that solved then?

**Comment by RoelandSalij** [comment]
It depends if properties are changed technically in the properties.xml (formatstring.xml in this widget).
A few versions back the display string was of type string, and we have changed it to translatable string. This causes the 'update all widgets' in this case.

As long as no xml properties were harmed, there is no issue. I agree the Business Modeler should handle a string to translatablestring transition more intelligently.

**Comment by pimvdnoll** [comment]
Hi Roeland thanks again for the quick response, I partly understand but I suggest if the widget is changed dramatically and no backward compatibility is possible make a new widget and fix this widget only for Mx 5.11 and higher please. In that case it's much better to replace the widget in fases.

Every developer who upgrades to (in my case 5.13.1) gets this problem because the formatstring is not working anymore.  So I would love to see only a fix/patch for Mx 5.13.1 in the previous version.

**Comment by RoelandSalij** [comment]
This change is there for several months, so reverting the change would cause the same issue with people using the previous version.

I agree it doesn't deserve a beauty award and we will definitely take it into consideration in future.

To provide you with a workaround, for now you can change line 113 in formatstring.xml(https://github.com/mendix/FormatString/blob/master/src/formatstring/formatstring.xml):
```xml
<property key="displaystr" type="translatableString" multiline="true">
```
to
```xml
<property key="displaystr" type="String" multiline="true">
```
